rust/sample-plugin/src/main.rs

```diff
@@ -33,7 +33,7 @@ use xi_plugin_lib::{Plugin, ChunkCache, View, mainloop, Error};
 /// intended to demonstrate how to edit a document; when the plugin is active,
 /// and the user inserts an exclamation mark, the plugin will capitalize the
 /// preceding word.
-struct SamplePlugin;
+struct SamplePlugin(usize);
```

Comment:
I'm not sure I'd like to include this update to the sample plugin in this PR. I'm not exactly sure what the role of the `sample-plugin` project is, and I do imagine updating that project to show examples of various plugin features, but this particular addition doesn't feel like it is a very useful illustration.

What might make the most sense for things like this, where we want some simple plugin just to demo a new feature, would be to copy `sample-plugin` into a new folder (out side of `xi-editor`) and then make a new repo for your new example, put that on your github, or something like this?